lib/kafo_parsers/puppet_strings_module_parser.rb

```diff
@@ -125,7 +129,11 @@ def sanitize(value)
     end
 
     def merge_doc_types(types)
-      Hash[@parsed_hash['signatures'].flatten.map { |param| [ param['name'], param['type'] ] }].merge(types)
+      Hash[tag_params.select { |param| !param['types'].nil? }.map { |param| [ param['name'], param['types'].first ] }].merge(types)
+    end
+
+    def tag_params
+      @parsed_hash['docstring']['tags'].select { |tag| tag['tag_name'] == 'param' }
```

Comment:
> `tag_params': private method `select' called for "tags":String (NoMethodError)

It'd appear that docstring is a string in your JSON output, not a hash (should contain 'text' and 'tags' keys), perhaps re-check that the version of puppet-strings being executed is definitely 0.99.0.

> I notice puppetlabs-strings is already 0.4.0, it seems the gem is falling behind

The module (puppet_labs_) isn't maintained any more, the gem will be more recent at 0.99.0. Perhaps check you don't have the module installed and taking precedence.